src/com/sahokia/kos/concerttickets/model/Ticket.java

```diff
@@ -73,52 +78,102 @@ public Ticket(String concertHall, String eventCode, boolean isPromo,
         }
     }
 
-    public String getId() {
-        return id;
-    }
-
+    @Override
     public void setId(String id) {
         if (id.length() <= 4) {
-            this.id = id;
+            super.setId(id);
         } else {
             throw new IllegalArgumentException("ID can not be longer than 4 characters");
         }
     }
 
-    public String getConcertHall() {
-        return concertHall;
+    @Override
+    public void print() {
+        System.out.println("ID of the ticket: " + this.getId());
```

Comment:
Seems like for printing we could override toString and reuse it :)

Reply:
Reused "toString" method in "print" method